### DISCUSSION TITLE
Automatic conversion of parameters from type Boolean to type enumeration

### DISCUSSION THREAD

**Discussion by AHaumer** [modelica/ModelicaStandardLibrary#3890] [Q&A] [answered]
Assume we have a `parameter Boolean b`, and we want to add more choices than `true|false`. Of course we would use an enumeration: `parameter MyChoices myChoice`
```
type MyChoices = enumeration(
  choice1 "former false", 
  choice2 "former true", 
  choice3 "new choice");
```
Of course we couly use a work-around:
```
parameter Boolean b;
parameter MyChoices myChoice=if b then MyChoices.choice1 else MyChoices.choice2;
```
The user can specify `b` and ignore `myChoice`, or ignore `b` and specify `myChoice`.
More elegant would be a conversion script converting `parameter Boolean b` to `parameter MyChoices myChoice`.
As far as I know this is'nt possible up to now. Are there any considerations solving this problem?

**Comment by HansOlsson** [comment, marked as answer]
> More elegant would be a conversion script converting `parameter Boolean b` to `parameter MyChoices myChoice`.
> As far as I know this is'nt possible up to now. Are there any considerations solving this problem?

To me it would seem that the following works:
`convertModifiers("Class", {"b"}, {"myChoice=if %b% then .MyChoices.choice1 else .MyChoice.choice2"}, true);`
However, it doesn't work for code _using_ the Boolean parameter.
